src/classes/BDI_FieldMappingCustomMetadata.cls

```diff
@@ -49,11 +70,65 @@ public with sharing class BDI_FieldMappingCustomMetadata implements BDI_FieldMap
     * TODO: Implement this method
     */
 
-    public Map<String,String> getFieldMap(String dataImportObjectName, 
-        String targetObjectName, List<String> dataImportFields) {
+    public Map<String,String> getFieldMap(String dataImportObjectName, String targetObjectName, List<String> dataImportFields) {
+
+        Map<String,String> dataImportFieldToTargetField = new Map<String,String>();
 
-        // TODO: Map fields based on CMT here
+        String ns = UTIL_Namespace.getNamespace();
+        //Retreive the Data Import Settings to determine the default field mapping set.
+        Data_Import_Settings__c dis = UTIL_CustomSettingsFacade.getDataImportSettings();
 
-        return new Map<String,String>();
+        String fieldMappingSetName = dis.Default_Data_Import_Field_Mapping_Set__c;
+        Data_Import_Field_Mapping_Set__mdt diFieldMappingSet;
+        Data_Import_Field_Mapping__mdt[]  diFieldMappings;
+        Data_Import_Object_Mapping__mdt diObjectMapping;
+
+        if(fieldMappingSetName != null){
+            diFieldMappingSet = [SELECT id, DeveloperName, 
+                                    Data_Import_Object_Mapping_Set__c,
+                                    Data_Import_Object_Mapping_Set__r.DeveloperName    
+                                FROM Data_Import_Field_Mapping_Set__mdt
+                                WHERE DeveloperName =: fieldMappingSetName LIMIT 1];
+
+            if(diFieldMappingSet != null && diFieldMappingSet.Data_Import_Object_Mapping_Set__c != null){
```

Comment:
It looks like there's something weird happening with tab alignment here? The "if()'s" keep creeping over to the right